test/integration/binary/binaries_test_fixture.hpp

```diff
@@ -0,0 +1,139 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef IROHA_BINARIES_TEST_FIXTURE_HPP
+#define IROHA_BINARIES_TEST_FIXTURE_HPP
+
+#include <gtest/gtest.h>
+
+#include <vector>
+#include "builders/protobuf/block.hpp"
+#include "framework/integration_framework/integration_test_framework.hpp"
+#include "framework/specified_visitor.hpp"
+#include "launchers.hpp"
+
+using namespace boost::process;
+using namespace binary_test;
+using namespace integration_framework;
+using namespace shared_model;
+using namespace shared_model::interface;
+
+namespace shared_model {
+
+  namespace proto {
+    class Query;
+  }
+
+  namespace interface {
+
+    class Block;
+    class AccountDetailResponse;
+    class AccountAssetResponse;
+    class AccountResponse;
+    class AssetResponse;
+    class RolePermissionsResponse;
+    class RolesResponse;
+    class SignatoriesResponse;
+    class TransactionsResponse;
+
+  }  // namespace interface
+
+}  // namespace shared_model
+
+namespace query_validation {
+
+  namespace internal {
+
+    class Void {};
+
+  }  // namespace internal
+
+  using QueryIterator = std::vector<proto::Query>::iterator;
+
+  template <typename ResponseType>
+  inline void checkQueryResponse(
+      const shared_model::proto::QueryResponse &response) {
+    ASSERT_NO_THROW(boost::apply_visitor(
+        framework::SpecifiedVisitor<ResponseType>(), response.get()));
+  }
+
+  template <typename Head, typename... Tail>
+  inline void _validateQueries(QueryIterator it,
+                               QueryIterator end,
+                               IntegrationTestFramework &itf) {
+    if (it != end) {
+      itf.sendQuery(*it, checkQueryResponse<Head>);
+      _validateQueries<Tail...>(++it, end, itf);
+    }
+  }
+
+  template <>
+  inline void _validateQueries<internal::Void>(QueryIterator it,
+                                               QueryIterator end,
+                                               IntegrationTestFramework &itf){};
+
+  template <typename... Responses>
+  inline void validateQueries(QueryIterator it,
+                              QueryIterator end,
+                              IntegrationTestFramework &itf) {
+    _validateQueries<Responses..., internal::Void>(it, end, itf);
+  }
+
+}  // namespace query_validation
+
+template <typename T>
```

Comment:
Add explanation of T.

Reply:
or even better rename to `SdkLauncher`

Reply:
This may be inconvenient in the future when you put to template objects which are different from `Launcher` type.